grant-project/zcash-grant-system
# Better logging & add sentry to blockchain watcher

### What This Does

Adds sentry to the blockchain watcher node process. Also configures the JS projects to use `NODE_ENV` (or `SENTRY_ENVIRONMENT`, in the case of frontend, since staging uses the `NODE_ENV=production` configuration for webpack purposes.)

Also replaces all logging with configured `winston` logger. This gives us granular logging with timestamps, and allows us to add excessive debug logging without messing with production. You can set `LOG_LEVEL` in .env to get more logging.

### Steps to Test

1. Start blockchain watcher, confirm you get pretty logging
2. Go to the devtool and send {}, causing an error. Confirm it logs the error.
3. Add `LOG_LEVEL="debug"` to your config, and do 2 again. Confirm you get debug info.

## Screenshots

<img width="604" alt="screen shot 2019-01-18 at 2 42 47 pm" src="https://user-images.githubusercontent.com/649992/51409417-c2601380-1b2f-11e9-87c7-e78421499a3a.png">
